openmdao/core/vec_wrapper.py

```diff
@@ -1195,6 +1196,9 @@ def setup(self, parent_params_vec, params_dict, srcvec, my_params,
                                                             owned=False,
                                                             imag_val=imag_val)
 
+        self._cache_units()
```

Comment:
I would put an 'if self.deriv_units' around this since the cache is only used when self.deriv_units is true.